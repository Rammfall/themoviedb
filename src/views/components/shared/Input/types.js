import PropTypes from 'prop-types'

export const propTypes = {
  placeholder: PropTypes.string,
  validateStatus: PropTypes.oneOf([
    'success',
    'warning',
    'error',
    'validating',
    ''
  ]),
  validationMessage: PropTypes.string,
  type: PropTypes.oneOf(['password', 'text']),
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  onBlur: PropTypes.func.isRequired,
  name: PropTypes.string,
  prefix: PropTypes.node
}

export const defaultProps = {
  placeholder: undefined,
  validateStatus: undefined,
  validationMessage: undefined,
  type: 'text',
  value: '',
  name: '',
  prefix: undefined
}
