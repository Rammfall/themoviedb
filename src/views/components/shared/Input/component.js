import React from 'react'
import PropTypes from 'prop-types'
import { Form, Input as AntInput } from 'antd'

const Input = ({
  placeholder,
  validateStatus,
  validationMessage,
  type,
  value,
  onChange,
  onBlur,
  name,
  prefix
}) => (
  <Form.Item
    validateStatus={validateStatus}
    help={validationMessage}
  >
    <AntInput
      placeholder={placeholder}
      type={type}
      onBlur={onBlur}
      prefix={prefix}
      onChange={onChange}
      name={name}
      value={value}
    />
  </Form.Item>
)

Input.propTypes = {
  placeholder: PropTypes.string,
  validateStatus: PropTypes.oneOf([
    'success',
    'warning',
    'error',
    'validating',
    ''
  ]),
  validationMessage: PropTypes.string,
  type: PropTypes.oneOf(['password', 'text']),
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  onBlur: PropTypes.func.isRequired,
  name: PropTypes.string,
  prefix: PropTypes.node
}

Input.defaultProps = {
  placeholder: undefined,
  validateStatus: undefined,
  validationMessage: undefined,
  type: 'text',
  value: '',
  name: '',
  prefix: undefined
}

export default Input
