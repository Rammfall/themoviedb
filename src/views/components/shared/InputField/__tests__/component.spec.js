import React from 'react'
import { shallow } from 'enzyme'

import Input from '../component'

describe('Input component matches snapshot', () => {
  const defaultProps = {
    field: {
      name: 'test',
      onChange: jest.fn(),
      onBlur: jest.fn(),
      value: 'test'
    },
    form: {
      touched: { test: true },
      errors: { test: 'test' }
    },
    placeholder: 'Test'
  }

  it('with default props', () => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    const component = shallow(<Input {...defaultProps} />)

    expect(component).toMatchSnapshot()
  })
})
