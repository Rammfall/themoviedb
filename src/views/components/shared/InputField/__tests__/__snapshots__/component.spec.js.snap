// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component matches snapshot matches snapshot 1`] = `
<Input
  name="test"
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  placeholder="Test"
  type="text"
  validateStatus="error"
  validationMessage="test"
  value="test"
/>
`;
